Drop-In/src/main/java/com/braintreepayments/api/dropin/DropInActivity.java

```diff
@@ -150,7 +160,7 @@ public void onResponse(String deviceData) {
             @Override
             public void onResponse(Boolean isReadyToPay) {
                 mSupportedPaymentMethodListView.setAdapter(new SupportedPaymentMethodsAdapter(
-                        DropInActivity.this, configuration, isReadyToPay, mClientTokenPresent,
+                        DropInActivity.this, configuration, isReadyToPay && mDropInRequest.isAndroidPayEnabled(), mClientTokenPresent,
```

Comment:
you have a rather fair point about _wating_. I just wanted to avoid branching for setting the adapter.